plugins/c8y_firmware_plugin/src/firmware_manager.rs

```diff
@@ -0,0 +1,734 @@
+        info!("Firmware update response received. Details: id={operation_id}, child={child_id}, status={status:?}");
+
+        let status_file_path = PersistentStore::get_file_path(operation_id);
+        if let Err(err) = PersistentStore::has_expected_permission(operation_id) {
+            warn!("{err}");
+            return Ok(vec![]);
+        }
+
+        let operation_entry = FirmwareOperationEntry::read_from_file(status_file_path.as_path())?;
+
+        let current_operation_state = self
+            .operation_timer
+            .current_value(&(child_id.to_string(), operation_id.to_string()));
+
+        let mut mapped_responses = vec![];
+        match current_operation_state {
+            Some(&ActiveOperationState::Executing) => {}
+            Some(&ActiveOperationState::Pending) => {
+                let executing_status_payload = DownloadFirmwareStatusMessage::status_executing()?;
+                mapped_responses.push(Message::new(&c8y_child_topic, executing_status_payload));
+            }
+            None => {
+                info!("Received a response with mismatched info. Ignore this response.");
+                return Ok(mapped_responses);
+            }
+        }
+
+        match status {
+            OperationStatus::Successful => {
+                self.operation_timer
+                    .stop_timer((child_id, operation_id.to_string()));
+                fs::remove_file(status_file_path)?;
+
+                let update_firmware_state_message = format!(
+                    "115,{},{},{}",
+                    operation_entry.name, operation_entry.version, operation_entry.server_url
+                );
+                mapped_responses.push(Message::new(
+                    &c8y_child_topic,
+                    update_firmware_state_message,
+                ));
+
+                let successful_status_payload =
+                    DownloadFirmwareStatusMessage::status_successful(None)?;
+                mapped_responses.push(Message::new(&c8y_child_topic, successful_status_payload));
+            }
+            OperationStatus::Failed => {
+                self.operation_timer
+                    .stop_timer((child_id, operation_id.to_string()));
+                fs::remove_file(status_file_path)?;
+
+                if let Some(error_message) = &child_device_payload.reason {
+                    let failed_status_payload =
+                        DownloadFirmwareStatusMessage::status_failed(error_message.clone())?;
+                    mapped_responses.push(Message::new(&c8y_child_topic, failed_status_payload));
+                } else {
+                    let default_error_message =
+                        String::from("No fail reason provided by child device.");
+                    let failed_status_payload =
+                        DownloadFirmwareStatusMessage::status_failed(default_error_message)?;
+                    mapped_responses.push(Message::new(&c8y_child_topic, failed_status_payload));
```

Comment:
```suggestion
                let failed_status_payload = DownloadFirmwareStatusMessage::status_failed(
                    child_device_payload
                        .reason
                        .unwrap_or_else(|| "No fail reason provided by child device.".into()),
                )?;
                mapped_responses.push(Message::new(&c8y_child_topic, failed_status_payload));
```

Reply:
I needed to change a bit as the suggested one cannot be complied.